platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,820 @@
+            redirectPage = IOUtils.toString(
+                    IdpEndpoint.class.getResourceAsStream("/templates/redirect.handlebars"));
+        } catch (Exception e) {
+            LOGGER.error("Unable to load index page for IDP.", e);
+        }
+    }
+
+    private void parseSpMetadata(List<String> spMetadata) {
+        if (spMetadata != null) {
+            try {
+                serviceProviders = MetadataConfigurationParser.buildEntityDescriptors(spMetadata);
+            } catch (IOException e) {
+                LOGGER.error("Unable to parse SP metadata configuration.", e);
+            }
+        }
+    }
+
+    @POST
+    public Response showPostLogin(@FormParam(SAML_REQ) String samlRequest,
+            @FormParam(RELAY_STATE) String relayState, @Context HttpServletRequest request,
+            @Context HttpServletResponse response) throws WSSecurityException {
+        LOGGER.debug("Recevied POST IdP request.");
+        return showLoginPage(samlRequest, relayState, null, null, request, true);
+    }
+
+    @GET
+    public Response showGetLogin(@QueryParam(SAML_REQ) String samlRequest,
+            @Encoded @QueryParam(RELAY_STATE) String relayState,
+            @QueryParam(SSOConstants.SIG_ALG) String signatureAlgorithm,
+            @QueryParam(SSOConstants.SIGNATURE) String signature,
+            @Context HttpServletRequest request, @Context HttpServletResponse response)
+            throws WSSecurityException {
+        LOGGER.debug("Recevied GET IdP request.");
+        return showLoginPage(samlRequest, relayState, signatureAlgorithm, signature, request, false);
+    }
+
+    private Response showLoginPage(String samlRequest, String relayState, String signatureAlgorithm,
+            String signature, HttpServletRequest request, boolean isPost)
+            throws WSSecurityException {
+        String responseStr;
+        AuthnRequest authnRequest = null;
+        try {
+            Map<String, Object> responseMap = new HashMap<>();
+            validateRelayState(relayState);
+            authnRequest = unwrapAuthnRequest(samlRequest, isPost);
+            validateAuthnRequest(authnRequest, samlRequest, relayState, signatureAlgorithm,
+                    signature, isPost);
+            X509Certificate[] certs = (X509Certificate[]) request
+                    .getAttribute("javax.servlet.request.X509Certificate");
+            boolean hasCerts = (certs != null && certs.length > 0);
+            if (authnRequest.isPassive() && hasCerts) {
+                LOGGER.debug("Received Passive & PKI AuthnRequest.");
+                org.opensaml.saml2.core.Response samlpResponse;
+                try {
+                    samlpResponse = handleLogin(authnRequest, PKI, request,
+                            authnRequest.isPassive());
+                    LOGGER.debug("Passive & PKI AuthnRequest logged in successfully.");
+                } catch (SecurityServiceException e) {
+                    LOGGER.error("", e);
```

Comment:
Do we want messages in these logs or just the exceptions?

And regarding the status codes returned, I assume this is relatively safe as it's system-to-system for a passive login, but it seems weird to me to respond with specific security failures. That's definitely a worry with an interactive system - letting an attacker know if a username exists by differentiating between bad pass and generic login exceptions being the prime example.